torch/csrc/inductor/inductor_ops.cpp

```diff
@@ -38,13 +39,33 @@ Tensor _reinterpret_tensor(
   return self_;
 }
 
+static void accumulate_grad_(const Tensor& variable, const Tensor& new_grad) {
+  at::Tensor& grad = variable.mutable_grad();
+  if (new_grad.device() != kMeta) {
+    torch::autograd::AccumulateGrad::accumulateGrad(
+        variable,
+        grad,
+        new_grad,
+        1 /* num_expected_refs */,
+        [&grad](at::Tensor&& grad_update) { grad = std::move(grad_update); });
```

Comment:
If we have a lifecycle preserving analogue of std::reference_wrapper that raises on access, that would be nice, because then if this lambda is stored or used after grad goes out of scope, we don't get undefined behavior.

That being said, I'm kinda assuming that this is just invoked immediately based on how you wrote it?

Reply:
This is the exact same thing accumulate_grad nodes do:
https://github.com/pytorch/pytorch/blob/d7317d8a1172dce9871803ff4b2d92d1acf869d6/torch/csrc/autograd/functions/accumulate_grad.cpp#L58